### PR TITLE
AMQP-462: Batching and Compression Docs

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/AMQP-462

- Batching
- Compression

### CODE REVIEW

File: src/reference/asciidoc/amqp.adoc [hunk 1]
```diff
@@ -715,9 +716,57 @@ This allows the sender to correlate a confirm (ack or nack) with the sent messag
...
+
+	/**
+	 * @return the date the next scheduled release should run, or null if no data to release.
+	 */
+	Date nextRelease();
+
+	/**
+	 * Release batch(es), perhaps due to a timeout.
+	 * @return The batched message(s).
+	 */
```
Comment (artembilan) [inline src/reference/asciidoc/amqp.adoc]: Not sure that we should share JavaDocs in Reference Manual.
Simple interface declaration should be enough

File: src/reference/asciidoc/amqp.adoc [hunk 2]
```diff
@@ -715,9 +716,57 @@ This allows the sender to correlate a confirm (ack or nack) with the sent messag
...
 
-Starting with *version 1.4* the `RabbitTemplate` supports the SpEL `mandatoryExpression` property, which is evaluated against each request message, as the root evaluation object, resolving to a `boolean` value.
+Starting with _version 1.4_ the `RabbitTemplate` supports the SpEL `mandatoryExpression` property, which is evaluated against each request message, as the root evaluation object, resolving to a `boolean` value.
 Bean references, such as `"@myBean.isMandatory(#root)"` can be used in the expression.
 
+[[template-batching]]
+===== Batching
+
+Starting with _version 1.4.2_, the `BatchingRabbitTemplate` has been introduced.
+This is a subclass of `RabbitTemplate` with an overridden `send` method that batches messages according to the `BatchingStrategy`; only when a batch is complete is the message sent to rabbitmq.
```
Comment (artembilan) [inline src/reference/asciidoc/amqp.adoc]: s/rabbitmq/RabbitMQ

Not sure that we should stop honoring our vendors :smile:

**Comment by artembilan** [comment]
Otherwise LGTM and can be fixed on merge.

**Comment by garyrussell** [comment]
ok; all sounds good; thanks

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/fba00a016abf3c212266762ebda9fc3620d486ae

**Pull request closed without merging**